TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Alerts/CloseAlert/Index.cshtml.cs

```diff
@@ -0,0 +1,74 @@
+using System.ComponentModel.DataAnnotations;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using TeachingRecordSystem.Core.Dqt.Queries;
+
+namespace TeachingRecordSystem.SupportUi.Pages.Alerts.CloseAlert;
+
+public class IndexModel : PageModel
+{
+    private readonly TrsLinkGenerator _linkGenerator;
+    private readonly ICrmQueryDispatcher _crmQueryDispatcher;
+
+    public IndexModel(
+        TrsLinkGenerator linkGenerator,
+        ICrmQueryDispatcher crmQueryDispatcher)
+    {
+        _linkGenerator = linkGenerator;
+        _crmQueryDispatcher = crmQueryDispatcher;
+    }
+
+    [FromRoute]
+    public Guid AlertId { get; set; }
+
+    [BindProperty(SupportsGet = true)]
+    [Display(Name = "End date")]
+    public DateOnly? EndDate { get; set; }
+
+    public Guid? PersonId { get; set; }
+
+    public async Task<IActionResult> OnGet()
+    {
+        var alert = await _crmQueryDispatcher.ExecuteQuery(new GetSanctionDetailsBySanctionIdQuery(AlertId));
```

Comment:
As with Confirm, could move this into `OnPageHandlerExecutionAsync`

Reply:
done